services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSpec.groovy

```diff
@@ -0,0 +1,601 @@
+        def srcSwitchCreatedCookies = getCookiesWithMeter(srcSwitch.dpId)
+        srcSwitchValidateInfo.meters.missing*.meterId[0] == newMeterId
+        srcSwitchValidateInfo.meters.missing*.cookie.containsAll(srcSwitchCreatedCookies)
+
+        srcSwitchValidateInfo.meters.missing.each {
+            assert it.rate == flow.maximumBandwidth
+            assert it.flowId == flow.id
+            assert ["KBPS", "BURST", "STATS"].containsAll(it.flags)
+            verifyBurstSizeIsCorrect(srcSwitchBurstSize, it.burstSize)
+        }
+
+        def dstSwitchCreatedCookies = getCookiesWithMeter(dstSwitch.dpId)
+        dstSwitchValidateInfo.meters.missing*.meterId[0] == newMeterId
+        dstSwitchValidateInfo.meters.missing*.cookie.containsAll(dstSwitchCreatedCookies)
+
+        dstSwitchValidateInfo.meters.missing.each {
+            assert it.rate == flow.maximumBandwidth
+            assert it.flowId == flow.id
+            assert ["KBPS", "BURST", "STATS"].containsAll(it.flags)
+            verifyBurstSizeIsCorrect(dstSwitchBurstSize, it.burstSize)
+        }
+
+        and: "The rest fields are empty on the src and dst switches"
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            verifyMeterSectionsAreEmpty(it, ["misconfigured", "proper"])
+        }
+
+        and: "Rules still exist in the 'proper' section on the src and dst switches"
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            verifyRuleSectionsAreEmpty(it, ["missing", "excess"])
+        }
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        and: "Delete excess meters on the src and dst switches"
+        srcSwitchCreatedMeterIds.each { northbound.deleteMeter(srcSwitch.dpId, it) }
+        dstSwitchCreatedMeterIds.each { northbound.deleteMeter(dstSwitch.dpId, it) }
+
+        then: "Check that the switch validate request returns empty sections"
+        Wrappers.wait(WAIT_OFFSET) {
+            def srcSwitchValidateInfoAfterDelete = northbound.switchValidate(srcSwitch.dpId)
+            def dstSwitchValidateInfoAfterDelete = northbound.switchValidate(dstSwitch.dpId)
+            [srcSwitchValidateInfoAfterDelete, dstSwitchValidateInfoAfterDelete].each {
+                verifyRuleSectionsAreEmpty(it)
+                verifyMeterSectionsAreEmpty(it)
+            }
+        }
+    }
+
+
+    def "Able to get empty switch validate information from the intermediate switch(flow contain > 2 switches)"() {
+        when: "Create an intermediate-switch flow"
+        def flow = intermediateSwitchFlow()
+        flowHelper.addFlow(flow)
+        def flowPath = PathHelper.convert(northbound.getFlowPath(flow.id))
+
+        then: "The intermediate switch does not contain any information about meter"
+        def intermediateSwitchValidateInfo = northbound.switchValidate(flowPath[1].switchId)
+        verifyMeterSectionsAreEmpty(intermediateSwitchValidateInfo)
```

Comment:
You can verify rules info for intermediate switches in this test as well, in order to save time and space.

Reply:
added
```
and: "Rules are stored in the 'proper' section on the transit switch"
intermediateSwitchValidateInfo.rules.proper.size() == 2
verifyRuleSectionsAreEmpty(intermediateSwitchValidateInfo, ["missing", "excess"])
```